node/modules/dtypes/scorekeeper.go

```diff
@@ -0,0 +1,24 @@
+package dtypes
+
+import (
+	"sync"
+
+	peer "github.com/libp2p/go-libp2p-core/peer"
+)
+
+type ScoreKeeper struct {
+	lk     sync.Mutex
+	scores map[peer.ID]float64
```

Comment:
We probably should have some size limit on this, even if set to a very large value

Reply:
It is just a copy of scores table from pubsub.

Reply:
There should be a size limit but it should be in pubsub.